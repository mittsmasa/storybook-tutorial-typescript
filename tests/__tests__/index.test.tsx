jest.mock("swr");
import React from "react";
import { mount } from "enzyme";
import Home from "../../pages";
import toJson from "enzyme-to-json";
import useSWR from "swr";

describe("index.tsx", () => {
  it("render", () => {
    (useSWR as jest.Mock).mockReturnValue({ data: "hoge" });

    const wrapper = mount(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
