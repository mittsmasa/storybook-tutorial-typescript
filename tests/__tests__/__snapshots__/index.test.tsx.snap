// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.tsx render 1`] = `
<Home>
  <div
    className="jsx-1914448416 "
  >
    <Head>
      <_default
        headManager={Object {}}
        inAmpMode={false}
        reduceComponentsToState={[Function]}
      />
    </Head>
    <main
      className="jsx-1914448416 "
    >
      <h1
        className="jsx-1914448416 "
      >
        Welcome to 
        <a
          className="jsx-1914448416"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        className="jsx-1914448416 "
      >
        Get started by editing 
        <code
          className="jsx-1914448416 "
        >
          pages/index.js
        </code>
      </p>
      <div
        className="jsx-1914448416 "
      >
        <a
          className="jsx-1914448416 "
          href="https://nextjs.org/docs"
        >
          <h2
            className="jsx-1914448416"
          >
            Documentation →
          </h2>
          <p
            className="jsx-1914448416"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          className="jsx-1914448416 "
          href="https://nextjs.org/learn"
        >
          <h2
            className="jsx-1914448416"
          >
            Learn →
          </h2>
          <p
            className="jsx-1914448416"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          className="jsx-1914448416 "
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h2
            className="jsx-1914448416"
          >
            Examples →
          </h2>
          <p
            className="jsx-1914448416"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          className="jsx-1914448416 "
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h2
            className="jsx-1914448416"
          >
            Deploy →
          </h2>
          <p
            className="jsx-1914448416"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      className="jsx-1914448416 "
    >
      <a
        className="jsx-1914448416"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by
         
        <span
          className="jsx-1914448416 "
        >
          <Image
            alt="Vercel Logo"
            height={16}
            src="/vercel.svg"
            width={72}
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzIiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style={
                    Object {
                      "border": "none",
                      "display": "block",
                      "margin": 0,
                      "maxWidth": "100%",
                      "padding": 0,
                    }
                  }
                />
              </div>
              <noscript>
                <img
                  alt="Vercel Logo"
                  decoding="async"
                  src="/_next/image?url=%2Fvercel.svg&w=256&q=75"
                  srcSet="/_next/image?url=%2Fvercel.svg&w=96&q=75 1x, /_next/image?url=%2Fvercel.svg&w=256&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
              <img
                alt="Vercel Logo"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </div>
          </Image>
        </span>
      </a>
    </footer>
    <JSXStyle
      id="1914448416"
    />
  </div>
</Home>
`;
